src/openforms/formio/components/custom.py

```diff
@@ -639,6 +644,127 @@ def as_json_schema(component: AddressNLComponent) -> JSONObject:
         return base
 
 
+class PartnerSerializer(serializers.Serializer):
+    bsn = serializers.CharField(
+        label=_("bsn"),
+        max_length=9,
+        help_text=_("The BSN of the partner"),
+        validators=[BSNValidator()],
+    )
+    initials = serializers.CharField(
+        label=_("initials"),
+        help_text=_("The initials of the partner"),
+        required=False,
+        allow_blank=True,
+    )
+    affixes = serializers.CharField(
+        label=_("affixes"),
+        help_text=_("The affixes of the partner"),
+        required=False,
+        allow_blank=True,
+    )
+    lastName = serializers.CharField(
+        label=_("last name"),
+        help_text=_("The last name of the partner"),
+        required=False,
+        allow_blank=True,
+    )
+    dateOfBirth = serializers.DateField(
+        label=_("date of birth"),
+        help_text=_("The date of birth of the partner"),
+        required=False,
+    )
+
+    def __init__(self, **kwargs):
+        self.component = kwargs.pop("component", None)
+        super().__init__(**kwargs)
+
+
+class PartnerListField(serializers.Field):
+    def __init__(self, component, **kwargs):
+        self.component = component
+        super().__init__(**kwargs)
+
+    def to_internal_value(self, data):
+        if not isinstance(data, list):
+            raise serializers.ValidationError("Expected a list of partners.")
+
+        serializer = PartnerSerializer(
+            data=data,
+            many=True,
+            component=self.component,
+        )
+        serializer.is_valid(raise_exception=True)
+
+        validated = serializer.validated_data
+        self.validate_list(validated)
+        return validated
+
+    def to_representation(self, value):
+        return PartnerSerializer(value, many=True, component=self.component).data
+
+    def validate_list(self, partners):
+        component_key = self.component["key"]
+        submission = self.context["submission"]
+        prefill_data = submission.get_prefilled_data()
+
+        fm_immutable_variable = FormVariable.objects.filter(
+            source=FormVariableSources.user_defined,
+            prefill_plugin=FM_PLUGIN_IDENTIFIER,
+            prefill_options__mutable_data_form_variable=component_key,
+        ).first()
+        if fm_immutable_variable:
+            # we do not send these information from the frontend (since they are not used
+            # until for now) so we have to exclude them from the data that needs validation
```

Comment:
```suggestion
            # we do not receive these fields from the frontend (since they are not used
            # for now) so we have to exclude them from the data that needs validation
```
Is this what you mean?